service/config_db.hpp

```diff
@@ -0,0 +1,262 @@
+//
+// Created by riz on 29/01/19.
+//
+
+#pragma once
+
+#include <cstring>
+#include <vector>
+#include <sstream>
+#include <sqlite_modern_cpp.h>
+
+namespace raven::db
+{
+  template<typename T>
+  class blob
+  {
+  public:
+    //! getters
+    template<typename vtype>
+    std::enable_if_t<std::is_fundamental<vtype>::value, vtype> get_value() const noexcept
+    {
+        vtype ret;
+        std::memcpy(&ret, data_.data(), sizeof(vtype));
+        return ret;
+    }
+
+    // le compilo me laisse pas faire une template specifier pour std::string je sais pas trop pourquoi ( je suis surement un abrutit mais la il est tard je suis fatigué) à investiguer
+    std::string get_value_string() const noexcept
+    {
+        return (std::string(data_.begin(), data_.end()));
+    }
+
+    const std::vector<T> &get_data() const noexcept
+    {
+        return data_;
+    }
+
+    std::vector<T> &get_data() noexcept
+    {
+        return data_;
+    }
+
+    //! Constructors
+    template<typename vtype, typename std::enable_if_t<std::is_fundamental<T>::value, int> = 0>
+    explicit blob(vtype value) noexcept
```

Comment:
Shouldn't this be taken by reference ?

Reply:
no because this function should be enabled only when `vtype` is a fundamental so taking by reference is not very relevant. (however I just notice thanks to this comment that I make a mistake in the line above it is `vtype` and not `T` which should be checked to be a fundamental or not)

Reply:
no because this function should be enabled only when vtype is a fundamental so taking by reference is not very relevant. (however I just notice thanks to this comment that I make a mistake in the line above it is vtype and not T which should be checked to be a fundamental or not)

Reply:
Right you are! Marking as resolved.